File: TypeScript/Mit Polymorphismus/Haustier.ts
export abstract class Haustier{
    protected name : string;

    constructor(n : string){
        this.name = n;
    }

    abstract fressen():void
    abstract freuen():void
    abstract lautGeben():void

    schreibeName(){
        console.log(this.name);
    }
}

File: C#/Mit Polymorphismus/Haustier.cs
using System;

public abstract class Haustier
{
	protected string name;

	public Haustier(string n)
    {
		name = n;
    }

	public abstract void Fressen();
	public abstract void Freuen();
	public abstract void LautGeben();

	public void SchreibeName()
    {
		System.Console.WriteLine(name);
    }
}

